fsh/ig-data/input/pagecontent/index.md

```diff
@@ -225,45 +226,42 @@ SMART app, and optionally navigates the user to a 'next' activity.
+| `activityType`       | CONDITIONAL | string | REQUIRED for `ui.launchActivity`, optional for `ui.done`. Navigation hint; see description below. |
+| `activityParameters` | CONDITIONAL | object | REQUIRED for `ui.launchActivity`, optional for `ui.done`. Navigation hint; see description below. |
 {:.grid}
 
 #### Response payload for `ui.done` and `ui.launchActivity`
-
-| Property              | Optionality  | Type   | Description |
-| --------------------- | ------------ | ------ | ----------- |
-| `success`        | REQUIRED  | boolean | `true` if the request has been accepted; `false` if it has been rejected. |
-| `details`        | OPTIONAL  | string | A human readable string explaining the outcome. |
+| Property  | Optionality | Type    | Description |
+| --------- | ----------- | ------- | ----------- |
+| `success` | REQUIRED    | boolean | `true` if the request has been accepted; `false` if it has been rejected. |
+| `details` | OPTIONAL    | string  | A human readable explanation of the outcome. |
 {:.grid}
 
-
 The `activityType` property conveys an activity type drawn from the SMART Web
-Messaging [Activity Catalog]. In general, these activities follow the same naming
-conventions as entries inthe CDS Hooks catalog (`noun-verb`), and will align with
-CDS Hooks catalog entries where feasible. The `activityType` property conveys a 
-navigation target such as `problem-add` or `order-sign`, indicating where EHR should
-go to after the ui message has been handled. An activity may specify additional parameters
-that can be included in the call as additional properties. 
+Messaging [Activity Catalog]. In general, these activities follow the same
+naming conventions as entries inthe CDS Hooks catalog (`noun-verb`), and will
+align with CDS Hooks catalog entries where feasible. The `activityType` property
+conveys a navigation target such as `problem-add` or `order-sign`, indicating
+where EHR should go to after the ui message has been handled. An activity may
+specify additional parameters that can be included in the call as additional
+properties.
 
-The `activityParameters` property conveys parameters specific to an activity type. See the
-SMART Web Messaging [Activity Catalog] for details.
+The `activityParameters` property conveys parameters specific to an activity
+type. See the SMART Web Messaging [Activity Catalog] for details.
 
 *Note:* A SMART app launched in the context of CDS Hooks should generally not
 need to specify an `activityType` or `activityParameters` with the `ui.done`
 message, because the EHR tracks the context in which the app was launched (e.g.,
-order entry) and can navigate to the appropriate follow-up screen based on this
+order entry) and can navigate to an appropriate follow-up screen based on this
 context.
 
-An example of a `ui.done` message is shown below:
+An example of a `ui.done` message from an app to the EHR is shown below:
 
 ```js
-appWindow.postMessage({
+window.postMessage({
```

Comment:
`window.postMessage` would mean sending a message to one's own window, because in the browser environment `window` is bound to the window holding the current DOM... which isn't what we want to demonstrate here.

Reply:
Ah, thanks for that.  I noticed that in the examples we were talking about having the app send messages to the EHR, but we were using `appWindow` as the variable name and that seemed backwards.  To match earlier examples, should the app examples continue using `targetWindow` as the variable name?

Reply:
I replaced all the `window` references in "app" code examples with `targetWindow` (resembling the original example).  I also replaced all `window.postMessage(..., clientAppOrigin)` references with `event.source.postMessage(..., event.origin)` for the EHR code examples.